core.js

```diff
@@ -104,6 +108,59 @@ export class FileTypeParser {
 		}
 	}
 
+	async toDetectionStream(webStream, options = {}) {
```

Comment:
I think this could be made simpler by using `pipeThrough`.


Example code (untested):


```js
function toDetectionStream(stream) {
	let buffer = new Uint8Array();
	let detectedFileType;

	const transformStream = new TransformStream({
		async start(controller) {
			const reader = stream.getReader();
			let bytesRead = 0;

			const {value: chunk, done} = await reader.read(new Uint8Array(sampleSize));
			if (done || !chunk) {
				detectedFileType = undefined;
			} else {
				try {
					detectedFileType = await this.fromBuffer(chunk.slice(0, sampleSize));
				} catch (error) {
					if (error instanceof strtok3.EndOfStreamError) {
						detectedFileType = undefined;
					} else {
						throw error;
					}
				}
			}

			controller.enqueue(value);
		},
		transform(chunk, controller) {
			controller.enqueue(chunk);
		}
	});

	const finalStream = stream.pipeThrough(transformStream);

	finalStream.fileType = detectedFileType;

	return finalStream;
}
```

Reply:
That would be great if that would work.
But when running `const finalStream = stream.pipeThrough(transformStream);`, it throws this error:
```
TypeError [ERR_INVALID_STATE]: Invalid state: The ReadableStream is locked
```
I tried to release the reader with `reader.releaseLock();`, in `start(controller)`, before the `pipeThrough`, but that did not resolve the issue.

See branch [try-determine-file-type-using-transform-stream](https://github.com/sindresorhus/file-type/tree/try-determine-file-type-using-transform-stream), commit 1bb56850263610585ccf0b13144d10f78fd78993.

Reply:
Got it working, by taking out the file-type detection of the transformation start(), to ensure the stream is unlocked before starting the pipeThrough(): 53ec046d56f496f41eed5f37f8b1faffaea2b428